sdk/tools/openrewrite-sample-compiler-ci.yml

```diff
@@ -0,0 +1,68 @@
+trigger:
+  branches:
+    include:
+      - main
+  paths:
+    include:
+      - /sdk/tools/azure-openrewrite/
+      - /sdk/tools/azure-openrewrite-compiler-maven-plugin/
+
+extends:
+  template: /eng/pipelines/templates/stages/1es-redirect.yml
+  parameters:
+    stages: 
+      - stage: CompileSamples
+        variables: 
+          - template: /eng/pipelines/templates/variables/globals.yml
+          - template: /eng/pipelines/templates/variables/image.yml
+
+        jobs:
+          - job: OpenRewriteCompilerPlugin
+            pool: 
+              name: $(LINUXPOOL)
+              image: $(LINUXVMIMAGE)
+              os: linux
```

Comment:
Do we only want to run this on linux?

Reply:
@srnagar I'm not aware of any particular reason to run duplicate the job on other operating systems. The logic of the compiler is simple enough that I can't imagine a scenario where we'd reasonably expect the result to differ by operating system.

Reply:
As discussed offline, we'll follow-up to add additional test images